graphbolt/include/csc_sampling_graph.h

```diff
@@ -0,0 +1,140 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file graphbolt/include/csc_sampling_graph.h
+ * @brief Header file of csc sampling graph.
+ */
+
+#include <torch/custom_class.h>
+#include <torch/torch.h>
+
+#include <string>
+#include <vector>
+
+namespace graphbolt {
+namespace sampling {
+
+using StringList = std::vector<std::string>;
+
+/**
+ * @brief Heterogeneous information in the Graph.
+ */
+struct HeteroInfo {
+  HeteroInfo(
+      const StringList& ntypes, const StringList& etypes,
+      torch::Tensor& node_type_offset, torch::Tensor& type_per_edge)
+      : node_types(ntypes),
+        edge_types(etypes),
+        node_type_offset(node_type_offset),
+        type_per_edge(type_per_edge) {}
+  /** @brief List of node types in the graph. */
+  StringList node_types;
+  /** @brief List of edge types in the graph. */
+  StringList edge_types;
+  /** @brief Offset array of node type. */
+  torch::Tensor node_type_offset;
+  /** @brief Type id of each edge, where type id is the corresponding index of
+   * edge_types. */
+  torch::Tensor type_per_edge;
+};
+
+/**
+ * @brief A sampling oriented csc format graph.
+ */
+class CSCSamplingGraph : public torch::CustomClassHolder {
+ public:
+  /**
+   * @brief Constructor for CSC with data.
+   * @param num_rows The number of rows in the dense shape.
+   * @param num_cols The number of columns in the dense shape.
+   * @param indptr The CSC format index pointer array.
+   * @param indices The CSC format index array.
+   * @param hetero_info Heterogeneous graph information, if present. Nullptr
+   * means it is a homogeneous graph.
+   */
+  CSCSamplingGraph(
+      int64_t num_rows, int64_t num_cols, torch::Tensor& indptr,
+      torch::Tensor& indices, const std::shared_ptr<HeteroInfo>& hetero_info);
+
+  /**
+   * @brief Create a CSC graph from tensors of CSC format.
+   * @param indptr Index pointer array of the CSC.
+   * @param indices Indices array of the CSC.
+   * @param shape Shape of the sparse matrix.
+   *
+   * @return CSCSamplingGraph
+   */
+  static c10::intrusive_ptr<CSCSamplingGraph> FromCSC(
+      const std::vector<int64_t>& shape, torch::Tensor indptr,
```

Comment:
Why this interface accepts a vector<int> while some of the previous one accepts two ints `num_rows` and `num_cols`?